pkg/graveler/ref/manager.go

```diff
@@ -84,15 +103,19 @@ func (m *KVManager) getRepositoryRec(ctx context.Context, repositoryID graveler.
 }
 
 func (m *KVManager) GetRepository(ctx context.Context, repositoryID graveler.RepositoryID) (*graveler.Repository, error) {
-	repo := graveler.RepositoryData{}
-	_, err := m.kvStore.GetMsg(ctx, graveler.RepositoriesPartition(), []byte(graveler.RepoPath(repositoryID)), &repo)
+	repo, err := m.getRepository(ctx, repositoryID)
 	if err != nil {
-		if errors.Is(err, kv.ErrNotFound) {
-			err = graveler.ErrRepositoryNotFound
-		}
 		return nil, err
 	}
-	return graveler.RepoFromProto(&repo).Repository, nil
+
+	switch repo.State {
+	case graveler.RepositoryStateActive:
+		return repo, nil
+	case graveler.RepositoryStateDeleted:
+		return nil, graveler.ErrRepositoryInDeletion
+	default:
```

Comment:
We have 1 more state. What about it?

Reply:
Currently we return the default error for state other than in-deletion and active.
This can happen for example when creating repository the second time while it is already being created, the chances of that are very small, thus I prefer not creating a new error type for it